### PR TITLE
Task on overlapping

### PR DESCRIPTION
**Pull request by yoavkt**
The entities task class is a pipeline class that enables the performance of a prediction task using a specific model.
Currently, the user can exclude certain entities from the task. Thus, the user could check the overlapping symbols of a task and a model and exclude the nonoverlapping. 

However, in some use cases, the user will want to compare several models on several tasks, so the user must manage an exclusion list per model. This change will enable the running of multiple models on the same task while creating an exclusion per model.
The following change will add this option to the run task script, allowing the user to use the workflow in one line. 

Currently, the code removes rows after encoding; for encodings that have a list of the covered symbols (such as pre-computed), we could have loaded the task with the covered genes only, yet for text encoders, we do not have the covered symbols for the more generic, yet maybe less effective solution was selected.

I'm uncertain about the summary. I left most of it on the original task, yet maybe we would like to report the task only on the overlap; what do you think?

### CODE REVIEW

File: gene_benchmark/tasks.py [hunk 1]
```diff
@@ -88,6 +88,19 @@ def _convert_df_to_3d_array(df: pd.DataFrame):
     return np.hstack(reshaped_embeddings)
 
 
+def _get_none_nan_instancies(
+    encodings: pd.DataFrame, outcomes: pd.Series | pd.DataFrame
+):
+    if encodings.shape[1] == 1:
+        nan_ind = encodings.squeeze().isna()
```
Comment (edenjenzohar) [inline gene_benchmark/tasks.py:95]: its hard to understand this logic, encodings.shape[1] == 1 , why does this need squeeze? 
cant we do this: 
```
# Find indices of rows with any NaN values
nan_rows = df.isna().any(axis=1)

# Get the indices of these rows
nan_indices = df.index[nan_rows]
```

File: gene_benchmark/tasks.py [hunk 2]
```diff
@@ -88,6 +88,19 @@ def _convert_df_to_3d_array(df: pd.DataFrame):
...
+    else:
+        nan_ind = encodings.isna().any()
+    if isinstance(outcomes, pd.Series):
+        return outcomes[~nan_ind]
+    else:
+        return outcomes.loc[~nan_ind, :]
+
```
Comment (edenjenzohar) [inline gene_benchmark/tasks.py:102]: also here do we really need this if statement? both series and dataframes support boolean indexing so this should work: 
`outcomes.loc[~nan_mask]`

**Review by edenjenzohar — COMMENTED** [review]
do we also have a test for when overlap is false? 
we need to check for both instances, and for each one, have tasks with genes that do overlap and don't.

File: gene_benchmark/tasks.py (hunk 2 above)
Comment (edenjenzohar) [reply gene_benchmark/tasks.py:102]: so in overall the function: 
`nan_rows = df.isna().any(axis=1)
 return outcomes.loc[~nan_rows]`

File: gene_benchmark/tasks.py (hunk 1 above)
Comment (yoavkt) [reply gene_benchmark/tasks.py:95]: `nan_rows = df.isna().any(axis=1)`
 solves it really nicely thanks for the syntax

File: gene_benchmark/tasks.py (hunk 2 above)
Comment (yoavkt) [reply gene_benchmark/tasks.py:102]: You are right

**Review by edenjenzohar — APPROVED** [review]

**Pull request merged**